source/robot_model/src/Model.cpp

```diff
@@ -192,10 +192,64 @@ StateRepresentation::CartesianPose Model::forward_geometry(const StateRepresenta
   return this->forward_geometry(joint_state, std::vector<std::string>{frame_name}).front();
 }
 
-StateRepresentation::JointPositions Model::inverse_geometry(const StateRepresentation::CartesianState&) const {
-  // TODO
-  return StateRepresentation::JointPositions();
-}
+StateRepresentation::JointPositions Model::inverse_geometry(StateRepresentation::JointState actual_joint_state,
+                                                              const StateRepresentation::CartesianState &desired_cartesian_state) {
+    // This method uses the angular velocity evaluated by the 'Infeasibility-free inverse kinematics method' which takes into account the
+    // joint limits. The angular velocity is then used for indicating the direction for an iterative algorithm.
+    // To converge at the desired end-effector position a similar approach to the Newton–Raphson method is used, the only difference is that
+    // here the pseudo-inverse of the jacobian is not used and instead is used the angualr velocity evaluated by the
+    // 'Infeasibility-free inverse kinematics method'.
+
+    const int JOINT_ID = this->robot_model_.njoints - 2;
+
+    // TODO if the actual_joint_state is not provided initialize the position of the robot in a radom conformation
+    // TODO take as input the parameters eps and IT_MAX
+    // TODO return 'success'
+
+    Eigen::VectorXd q = actual_joint_state.get_positions(); //pinocchio::neutral(model);
+    const double eps = 1e-4;
+    const int IT_MAX = 1000;
+
+    std::vector<unsigned int> frame_ids{(unsigned int)JOINT_ID};
```

Comment:
c-style casts should be avoided.

Also it seems unfortunate to have to make a vector with one element just to use the `forward_geometry` function. @buschbapti we should add overload to accept a single frame ID (private of course based on our recent discussion, since the user should use frame names instead)

There is clearly also some confusion about where the `JOINT_ID` comes from in the first place. I suggest you instead take the `const std::string& frame_name=""` as an input argument, and if empty you take the last element in the frame name list (`get_frame_names().back()` or something like that)

Reply:
ahhh that is a good option! Thanks.

Reply:
yes indeed and have the same function with a single `frame_name` as well

Reply:
Well check PR #49